### DISCUSSION TITLE
Does the undock keyboard button works on the SB1?

### DISCUSSION THREAD

**Discussion by RafaelSoaresP** [linux-surface/linux-surface#960] [Q&A] [answered]
Could find this specific info on previous discussions nor on the features matrix.

**Comment by qzed** [comment, marked as answer]
Yes. WIth some caveats.

The button is in large parts handled by the firmware, so it will work (within some limitations) with any OS. The detachment process normally sends an event to a software component (DTX daemon, missing for the SB1). After some timeout (which happens if the software is not present), it chooses one of two actions:
- If the latch is "locked" (can be set by software, e.g., to prevent detachment if the dGPU is in use), the LED will light up red and the detachment will be aborted, meaning you cannot detach the clipboard part. Normally the latch should however be "unlocked".
- If the latch is "unlocked" the release-mechanism will open and you can detach the clipboard.

In essence, this means: Normally everything should work. However, detachment will take longer as it's waiting for the timeout due to the missing driver/software component. Nevertheless, you will still be able to detach the clipboard. If you get a red light, try resetting the device (two-button reset) or boot into windows.

**Reply by NP-chaonay** [reply]
I am not using SB1 however does this mean that even Linux using dgpu this will still do an undocking (I know that consequences when dgpu is used but the base got undocked)?

**Reply by qzed** [reply]
Yes. Usage of the dGPU needs to be detected by software, there's no hardware detection. So it will happily allow you to undock while that is still in use.

**Reply by rexendevar** [reply]
is there a way to manually override the red light in software?
